docs/src/components/ComponentPlayground/propGenerators.tsx

```diff
@@ -0,0 +1,128 @@
+import { useBooleanKnob, useSelectKnob, useStringKnob } from '@stardust-ui/docs-components'
+import { Button } from '@stardust-ui/react'
+import * as _ from 'lodash'
+import * as faker from 'faker'
+import * as React from 'react'
+
+import { KnobGenerator } from 'docs/src/types'
+import { number } from 'docs/src/components/ComponentPlayground/typeGenerators'
+
+export const content: KnobGenerator<string> = ({ propName }) => ({
+  hook: useStringKnob,
+  name: propName,
+  initialValue: _.capitalize(`${faker.hacker.verb()} ${faker.hacker.noun()}`),
+})
+
+export const color: KnobGenerator<string> = ({ propName, propDef, componentInfo, theme }) => ({
+  hook: useSelectKnob,
+  name: propName,
+  allowsNone: true,
+  initialValue: propDef.defaultValue,
+  values: Object.keys({
+    ...theme.siteVariables.contextualColors,
+    ...theme.siteVariables.naturalColors,
+  }),
+})
+
+export const name: KnobGenerator<string> = ({ componentInfo, propDef, propName, theme }) => {
+  if (componentInfo.displayName === 'Icon') {
+    const values = Object.keys(theme.icons).slice(0, 10)
+
+    return {
+      hook: useSelectKnob,
+      name: propName,
+      allowsNone: _.isNil(propDef.defaultValue),
+      initialValue: propDef.defaultValue,
+      values,
+    }
+  }
+
+  if (componentInfo.displayName === 'Avatar') {
```

Comment:
💣 this code creates tight coupling between `propGenerators.tsx` and `Avatar.tsx`. Can we extract this in some `enhancers` file and have the logic there and consume `enhancers` here?

Reply:
For now I want to keep it dumb, even with this relationship because it's easy to understand why exact value was generated. Not all follow ups are implemented yet and don't see all cases, premature refactor there can lead into wrong direction.

Reply:
Yeah I would agree that this looks strange... Let's create a separate file to store all these mappings.

Reply:
Done, please check `componentGenerators.ts`